### DISCUSSION TITLE
Animation flicker with transition:leave

### DISCUSSION THREAD

**Discussion by garrytaulu** [alpinejs/alpine#2861] [7. Bugs (archived)] [answered]
I'm porting my personal website [garry.design](https://garry.design) from vanilla JS to Alpine and I'm running into an issue re-creating the functionality of the menu opening and closing. Currently I use vanilla to add classes and wait for animationend to remove classes. With Alpine when the menu-disappear class is added with transition:leave it flickers right at the end. 

I have a simple reproduction here [https://codepen.io/garrytaulu/pen/eYyXjEo](https://codepen.io/garrytaulu/pen/eYyXjEo) and a screen recording.

https://user-images.githubusercontent.com/730779/164484355-96333678-fddd-4567-a11a-c3047b215048.mov

.

**Comment by Tim-Wils** [comment, marked as answer]
Hi! You can use `animation-fill-mode: fowards` in css to prevent te flicker like so: https://codepen.io/tim-wils/pen/popBEvm

**Reply by SimoTod** [reply]
That's really clever. I think it should be added to the alpine directive as a default. I know a lot of people are having issues with x-transition and Safari

**Reply by garrytaulu** [reply]
Hi! Thanks for the reply, that solves my problem 😊. It's actually occurring on Chrome too but this definitely fixed it.

The Alpine implementation still isn't as smooth as my vanilla JS implementation (like when spam clicking the button) but I can at least move on with the rest of my site.

Edit: BTW `forwards` didn't seem to solve it in my actual implementation but `both` did.

**Reply by Tim-Wils** [reply]
Just looked into the source of the x-transition directive to see where this should be added, but I see it's based on css `transition` instead of `animation`.

I also tried to animate to 101% to stall the animation a little bit longer which also seems to work in Chrome for me.
```
@keyframes menu-disappear {
	0% {
		transform: scale(1);
	}
	60% {
		transform: scale(1.05);
		opacity: 1;
	}
	**101% {**
		opacity: 0;
		transform: scale(0);
	}
}
```
Looks like there's a very small timing thing going on when switching leave-classes at the end.

Guessing right now (cannot test it right now):
Maybe the timing from should (duration - 1) here and then wrapped in another animationframe on this line:
https://github.com/alpinejs/alpine/blob/7501305877dc7ed5ae12e29ca18efef60f438c73/packages/alpinejs/src/directives/x-transition.js#L297
Maybe this works:
`setTimeout(requestAnimationFrame(() => {el._x_transitioning.finish}), duration + delay)`
or this:
`setTimeout(requestAnimationFrame(() => {el._x_transitioning.finish}), (duration-1) + delay)`
